addons/web/controllers/webclient.py

```diff
@@ -148,3 +148,10 @@ def bundle(self, bundle_name, **bundle_params):
         } for tag, attrs, content in files]
 
         return request.make_json_response(data)
+
+    @http.route('/web/webclient/check_record_company', type='json', auth="user")
+    def check_record_company(self, model, res_id, company_ids):
+        company_dependant = 'company_id' in request.env[model]
+        if company_dependant:
+            return bool(request.env[model].sudo().search_count([('id', '=', res_id), ('company_id', 'in', company_ids)], limit=1))
```

Comment:
why not check_access_rights && check_access_rule, all of this without sudo ?

Reply:
I can't force company on them and that creates problems for me because I want to check_access_right for the next company not the current one

Reply:
did what WAN suggested !
Thanks